Client/Application/SceneDelegate.swift

```diff
@@ -0,0 +1,308 @@
+            browserViewController.openBlankNewTab(focusLocationField: false)
+        }
+
+        // If the `NSUserActivity` has a `webpageURL`, it is either a deep link or an old history item
+        // reached via a "Spotlight" search before we began indexing visited pages via CoreSpotlight.
+        if let url = userActivity.webpageURL {
+            let query = url.getQuery()
+
+            // Check for fxa sign-in code and launch the login screen directly
+            if query["signin"] != nil {
+                // bvc.launchFxAFromDeeplinkURL(url) // Was using Adjust. Consider hooking up again when replacement system in-place.
+            }
+
+            // Per Adjust documentation, https://docs.adjust.com/en/universal-links/#running-campaigns-through-universal-links,
+            // it is recommended that links contain the `deep_link` query parameter. This link will also
+            // be url encoded.
+            if let deepLink = query["deep_link"]?.removingPercentEncoding, let url = URL(string: deepLink) {
+                browserViewController.switchToTabForURLOrOpen(url)
+            }
+
+            browserViewController.switchToTabForURLOrOpen(url)
+        }
+
+        // Otherwise, check if the `NSUserActivity` is a CoreSpotlight item and switch to its tab or
+        // open a new one.
+        if userActivity.activityType == CSSearchableItemActionType {
+            if let userInfo = userActivity.userInfo,
+                let urlString = userInfo[CSSearchableItemActivityIdentifier] as? String,
+                let url = URL(string: urlString) {
+                browserViewController.switchToTabForURLOrOpen(url)
+            }
+        }
+
+    }
+
+    /// Use this method to notify the user that the specified activity couldn't be completed.
+    ///
+    /// We need to think about the right ways to notifiy users
+    func scene(
+        _ scene: UIScene,
+        didFailToContinueUserActivityWithType userActivityType: String, error: Error
+    ) {
+        // no-op
+    }
+
+    // MARK: - Responding to Scene (environment) Changes
+
+    /// Use this method to handle certain environment changes and adjust your scene accordingly.
+    ///
+    /// See how / what needs to be done here so that we can lock orientation on modal presentation as well.
+    func windowScene(
+        _ windowScene: UIWindowScene,
+        didUpdate previousCoordinateSpace: UICoordinateSpace,
+        interfaceOrientation previousInterfaceOrientation: UIInterfaceOrientation,
+        traitCollection previousTraitCollection: UITraitCollection
+    ) {
+        // no-op
+        // return self.orientationLock
+    }
+
+    // MARK: - Performing Tasks
+
+    /// Use this method to handle a selected shortcut action.
+    ///
+    /// Invoked when:
+    /// - a user activates the application by selecting a shortcut item on the home screen AND
+    /// - the window scene is already connected.
```

Comment:
I'm looking into a way around the second requirement.

Reply:
The way around it is by looking at connectionOptions in willConnectTo, and forward / handle deeplinks & shortcuts from there.